domaintools/utils.py

```diff
@@ -98,3 +99,15 @@ def prune_data(data_obj):
             if not isinstance(item, int) and not item:
                 items_to_prune.append(index)
         data_obj[:] = [item for index, item in enumerate(data_obj) if index not in items_to_prune and len(item)]
+
+
+def find_emails(data_str):
+    """Find and returns all emails"""
+    return set(re.findall(r'[\w\.-]+@[\w\.-]+', data_str))
+
+
+def find_ips(data_str):
+    """Find and returns all ipv4"""
+    ipv4s = set(re.findall(r'\b(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\b', data_str))
```

Comment:
Tried a bunch of regexes to pull out ipv6s but haven't found one that worked well for this so left those out for now.